src/communication/stream_transfer_sender.cpp

```diff
@@ -116,7 +151,8 @@ namespace cubstream
 				    cubstream::stream_position begin_sending_position)
     : m_channel (std::move (chn)),
       m_stream (stream),
-      m_last_sent_position (begin_sending_position)
+      m_last_sent_position (begin_sending_position),
+      m_is_termination_phase (false)
```

Comment:
Is related to your changes,  but please initialize m_p_stream_ack member and move comma before the member `, m_is_termination_phase`